Source/JavaScriptCore/wasm/WasmSectionParser.cpp

```diff
@@ -730,8 +760,8 @@ auto SectionParser::parseStructType(uint32_t position, RefPtr<TypeDefinition>& s
     WASM_PARSER_FAIL_IF(!fields.tryReserveCapacity(fieldCount), "can't allocate enough memory for struct fields ", fieldCount, " entries");
 
     for (uint32_t fieldIndex = 0; fieldIndex < fieldCount; ++fieldIndex) {
-        Type fieldType;
-        WASM_PARSER_FAIL_IF(!parseValueType(m_info, fieldType), "can't get ", fieldIndex, "th field Type");
+        StorageType fieldType;
+        WASM_PARSER_FAIL_IF(!parseStorageType(fieldType), "can't get ", fieldIndex, "th field Type");
```

Comment:
We may want to consider disallowing packed typed in structs temporaily, with a `FIXME` comment for example. My reasoning was that the struct instructions patch that's currently being reviewed doesn't have support for packed types yet, so it may be best to delay it (also I think it's untested in this patch right?).